js/private/js_run_devserver.mjs

```diff
@@ -143,6 +143,114 @@ function partitionArray(array, callback) {
     )
 }
 
+async function syncSymlink(src, dst, sandbox, exists) {
```

Comment:
These 3 `sync{Symlink,Directory,File}` methods were moved as-is except where I add additional comments on the PR